File: runme/mcgen/local/datan/exasrc/e1reg.f
      PROGRAM E1REG
      IMPLICIT DOUBLE PRECISION (A-H,O-Z)
      PARAMETER(N=10,NR=10)
      DIMENSION T(N),Y(N),DELTAY(N),X(NR),CHI2(NR)
      DIMENSION B(NR,NR),A(N,NR)
      PARAMETER (ZERO=0.D0,ONE=1.D0)
      DATA T
     +/-.9D0,-.7D0,-.5D0,-.3D0,-.1D0,.1D0,.3D0,.5D0,.7D0,.9D0/
      DATA Y /81.D0,50.D0,35.D0,27.D0,26.D0,
     +60.D0,106.D0,189.D0,318.D0,520.D0/
C identify program to user
      WRITE(*,*)' Program E1REG demonstrates use of REGPOL'
      WRITE(*,*)' '
C set errors
      DO 10 I=1,N
          DELTAY(I)=SQRT(Y(I))
10    CONTINUE
C perform polynomial regression
      CALL REGPOL(T,Y,DELTAY,N,NR,X,B,A,CHI2)
C output
      WRITE(*,*)' T '
      WRITE(*,'(10F7.2)')T
      WRITE(*,*)' Y'
      WRITE(*,'(10F7.2)')Y
      WRITE(*,*)' DELTAY'
      WRITE(*,'(10F7.2)')DELTAY
      WRITE(*,*)' X'
      WRITE(*,'(10F7.2)')X
      WRITE(*,*)' CHI2'
      WRITE(*,'(10F7.2)')CHI2
      END

File: runme/mcgen/local/datan/exasrc/e1reg.c
#include <stdio.h>
#include <math.h>
#include "datsrc.h"

/* Table of constant values */

static integer c__10 = 10;

main(void)
{
    /* Initialized data */ /* Local variables */
    static integer i;
    static double t[10] = { -.9,-.7,-.5,-.3,-.1,.1,.3,.5,.7,.9 },
                  y[10] = { 81.,50.,35.,27.,26.,60.,106.,189.,318.,520. },
                  a[100] /* was [10][10] */, b[100] /* was [10][10] */,
                  x[10], deltay[10], chi2[10];
/* identify program to user */
    printf("%s\n\n", "Program E1REG demonstrates use of REGPOL.");
/* set errors */
    for (i = 1; i <= 10; ++i) {
	deltay[i - 1] = sqrt(y[i - 1]);
    }
/* perform polynomial regression */
    regpol_(t, y, deltay, &c__10, &c__10, x, b, a, chi2);
/* output */
    printf("%s\n", "T");
    for (i = 1; i <= 10; ++i)
        printf("%7.2f", t[i-1]);
    printf("\n");
    printf("%s\n", "Y");
    for (i = 1; i <= 10; ++i)
        printf("%7.2f", y[i-1]);
    printf("\n");
    printf("%s\n", "DELTAY");
    for (i = 1; i <= 10; ++i)
        printf("%7.2f", deltay[i-1]);
    printf("\n");
    printf("%s\n", "X");
    for (i = 1; i <= 10; ++i)
        printf("%7.2f", x[i-1]);
    printf("\n");
    printf("%s\n", "CHI2");
    for (i = 1; i <= 10; ++i)
        printf("%7.2f", chi2[i-1]);
    printf("\n");
    return 0;
} /* main */

